### ISSUE TITLE
Program crashes on launch after Windows updates.

### ISSUE THREAD

**Issue by TheCongressman** [Ryochan7/DS4Windows#901]
After updating windows, I am getting this error message on program launch. I reverted the updates to verify the program was working. Now after updating again, I am getting the error.

![ds4windowserror](https://user-images.githubusercontent.com/58096800/69471662-b8222d80-0d6f-11ea-89f4-79a78c0806ae.jpg)

"The application was unable to start correctly (0xc0000135)."

**Comment by Ryochan7** [comment]
Very helpful Windows. That stinks. Luckily I have not installed the latest Windows 10 updates yet. Can you check if there is an error printed in the Windows Event Viewer; it is located at Control Panel > System and Security > Administrative Tools > Event Viewer. Check the Application logs and I think you will look for an item with a level of Error and a Source of .NET Runtime. Hopefully this type of problem will be easier to diagnose in the future as DS4Windows 2.0 will actually have logging support that will write out errors that have caused the application to crash.

Also, maybe you should check out the current WPF beta of DS4Windows 2.0. That particular build does not have logging yet though. I still have a couple of test to perform before I get another beta build out there.

https://drive.google.com/file/d/1Nu2gJhGx9ZDQ1gnQP1qU2LwgPqjVvTyj/view?usp=sharing

**Comment by Ryochan7** [comment]
Here is the newest WPF beta build. This build has logging support. Logs are stored in the Logs folder of your config directory.

https://drive.google.com/file/d/1VQE_ukZhIo-009qr1BF2IEo5x7Q5mcLo/view?usp=sharing

**Comment by mika-n** [comment]
What WindowsOS update caused this problem? Some major feature release or a specific fixpack? Win7 or Win10?

After installing the problematic update then make sure you have the latest version of ViGem driver, VisualC++ and .NETFramework runtime libraries also before trying DS4Win app.
https://github.com/Ryochan7/DS4Windows/wiki/Troubleshooting#link-list

DS4Windows self-updated doesn't update any of those runtime libaries or even ViGem driver if it happens to be some older version.

**Comment by Ryochan7** [comment]
He is probably referring to Windows 10 update 1909. I finally got it installed. Version 1.7.21 and 1.7.22 work just fine on my machine. However, the update did revert many things on my end like disabled devices and also controllers are appearing as new hardware again. I don't have it installed but I would bet update 1909 would have removed HidGuardian support.

**Comment by Ryusennin** [comment]
By the look of the UI, it's about Win10.

I have installed the latest Slow Ring update and didn't have any such bug, however as with every Win10 update so far I always have to reinstall the HidGuardian driver (and a truckload of other ones like Radeon, Realtek or my MIDI drivers). Thankfully Win7 is still my main system.

[Edit: Ryochan beat me to it.]

**Comment by TheCongressman** [comment]
Ok, it's been a wild couple of days. It seems as though 1909 update did a lot more than break DS4 functionality...

I looked at the event viewer as you asked, however that was also non-functional with a missing MCC View error. Which sent me on down a rabbit hole trying to resolve, which ultimately lead to the OS to no longer boot. I couldn't repair or revert, and ended up reinstalling. I am now on version 1903, which is working well.

I recently built a new pc and migrated the existing install over which I suspect lead to some of these more major issues. I will keep you posted if and when I update to the latest windows again.
